From fd019eca92a423e83e49a94f16ebca930e0fe51d Mon Sep 17 00:00:00 2001
From: Andreas Sturmlechner <asturm@gentoo.org>
Date: Fri, 23 Apr 2021 20:10:31 +0200
Subject: [PATCH] Move Qt5Concurrent to private link libraries, drop from
 KF5KIOConfig.cmake.in

This was added in 2f83dde2c8e8a25f0fe3dd52012bb49cb69e1ea5 but as far as I can
see it is only a KIO build time dependency with no Qt5Concurrent header being
used in KIO headers.

Test-built various revdeps successfully against the patched KIO with/without
Qt5Concurrent installed, fixing implicit deps while at it. KDE Gear packages
are fixed in >=21.04.2, KDE Plasma in >=5.21.5.

Signed-off-by: Andreas Sturmlechner <asturm@gentoo.org>
---
 KF5KIOConfig.cmake.in   | 1 -
 src/core/CMakeLists.txt | 2 +-
 2 files changed, 1 insertion(+), 2 deletions(-)

diff --git a/KF5KIOConfig.cmake.in b/KF5KIOConfig.cmake.in
index 687ec512..749e9b9b 100644
--- a/KF5KIOConfig.cmake.in
+++ b/KF5KIOConfig.cmake.in
@@ -21,7 +21,6 @@ find_dependency(KF5WindowSystem "@KF_DEP_VERSION@")
 endif()
 
 find_dependency(Qt5Network "@REQUIRED_QT_VERSION@")
-find_dependency(Qt5Concurrent "@REQUIRED_QT_VERSION@")
 find_dependency(Qt5DBus "@REQUIRED_QT_VERSION@")
 
 include("${CMAKE_CURRENT_LIST_DIR}/KF5KIOTargets.cmake")
diff --git a/src/core/CMakeLists.txt b/src/core/CMakeLists.txt
index b35258a7..16c83d59 100644
--- a/src/core/CMakeLists.txt
+++ b/src/core/CMakeLists.txt
@@ -201,9 +201,9 @@ PUBLIC
  KF5::CoreAddons   # KJob
  KF5::Service # TODO KF6 move to PRIVATE
  Qt5::Network
- Qt5::Concurrent            # QtConcurrentRun in hostinfo.cpp # TODO KF6 move to PRIVATE
  Qt5::DBus
 PRIVATE
+ Qt5::Concurrent            # QtConcurrentRun in hostinfo.cpp # TODO KF6 move to PRIVATE
  Qt5::Xml                   # davjob.cpp uses QDom
  KF5::ConfigCore
  KF5::I18n
-- 
2.32.0

